### PR TITLE
feat: 🎸 add selection effects to searchResult

### PR DESCRIPTION
**Pull request by dave-ok**
SearchResult component slides to right when clicked or selected in
state. Scroll to view when selected

BREAKING CHANGE: 🧨 None

✅ Closes: WIN-22

### CODE REVIEW

**Comment by dave-ok** [comment]
This adds some selection effects to the floating hotel search results. This does not touch the map and markers however
The component `searchResult` is selected when the `selectedFacilityId` state matches it's facilityId. 
This state should ideally be higher up in a shared state so that the markers and the `Results` component will always indicate the actually selected facility. I wasn't sure how far up this state should be shared, (maybe global) but will be updated appropriately

**Review by bobetbat — APPROVED** [review]
LGTM

File: src/components/Results.tsx [hunk 1]
```diff
@@ -117,6 +124,29 @@ export const Results: React.FC<{
     return null;
   }
 
+  const searchResultsRefs = useMemo(
+    () =>
+      filteredFacilities.reduce((refs, facility) => {
+        const ref = createRef<HTMLDivElement>();
+        return { ...refs, [facility.id]: ref };
+      }, {}),
+    []
```
Comment (kostysh) [inline src/components/Results.tsx:133]: I guess `filteredFacilities` must be added to the dependencies of this memoised constant.

File: src/components/Results.tsx [hunk 2]
```diff
@@ -117,6 +124,29 @@ export const Results: React.FC<{
...
+  );
+
+  const handleFacilitySelection = useCallback((facilityId: string) => {
```
Comment (kostysh) [inline src/components/Results.tsx:136]: In such a case `useCallback` is useless, you can simply define a regular callback

File: src/components/Results.tsx [hunk 3]
```diff
@@ -117,6 +124,29 @@ export const Results: React.FC<{
...
+    setSelectedFacilityId(facilityId);
+  }, []);
+
+  useEffect(() => {
+    // scroll to searchResult
+    selectedFaciltyId && searchResultsRefs[selectedFaciltyId]?.current?.scrollIntoView();
+  }, [selectedFaciltyId]);
```
Comment (kostysh) [inline src/components/Results.tsx:143]: `searchResultsRefs` must be added to the dependencies of `useEffect`

File: src/components/Results.tsx (hunk 1 above)
Comment (dave-ok) [reply src/components/Results.tsx:133]: Yes, you are right. Thank you. Will update this

**Comment by valera-rozuvan** [comment]
@dave-ok please resolve merge conflicts in file `src/components/Results.tsx`.

**Comment by dave-ok** [comment]
@kostysh Can you please review once more?

**Review by kostysh — APPROVED** [review]
LGTM

**Pull request merged**